gpii/node_modules/matchMakerFramework/src/MatchMakerUtilities.js

```diff
@@ -175,6 +188,135 @@ var fluid = fluid || require("infusion"),
         return togo;
     };
 
+    // true if one of the values matches across the two arrays (ignoring the indexing)
+    gpii.matchMakerFramework.utils.containsSameValue = function (arr1, arr2) {
+        var obj1 = {};
+        var i;
+        // convert arr1 to object for fast and easy comparison
+        for (i in arr1) {
+            obj1[arr1[i]] = true;
+        }
+        // compare:
+        for (i in arr2) {
+            if (arr2[i] in obj1) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    /**
```

Comment:
Add a proper doc comment here explaining the structure and contents of solrecs and data, what the return from this function is, and which data gets modified (a clearer model would have all the input data immutable and just accumulate fresh data into the return value)